src/gossip/table.zig

```diff
@@ -339,23 +339,37 @@ pub const GossipTable = struct {
         values: []SignedGossipData,
         timeout: u64,
         failed_indexes: *std.ArrayList(usize),
-    ) error{OutOfMemory}!void {
+    ) error{OutOfMemory}!struct {
+        timeout_count: u64,
+        success_count: u64,
+    } {
         failed_indexes.clearRetainingCapacity();
         try failed_indexes.ensureTotalCapacity(values.len);
 
+        var timeout_count: u64 = 0;
+        var success_count: u64 = 0;
+
         for (values, 0..) |value, index| {
             const value_time = value.wallclock();
             const is_too_new = value_time > now +| timeout;
             const is_too_old = value_time < now -| timeout;
             if (is_too_new or is_too_old) {
+                timeout_count += 1;
                 continue;
             }
 
             const did_insert = self.insert(value, now) catch false;
```

Comment:
I'm not sure about the design of the `insert` function.

this function returns only `true` or an error. 
`false` - is unreachable due to (line 274)
```zig
if (old_entry.value_hash.order(&versioned_value.value_hash) != .eq) {
    // if hash isnt the same and override() is false then msg is old
    try self.purged.insert(old_entry.value_hash, now);
    return InsertionError.OldValue;
} else {
    // hash is the same then its a duplicate
    return InsertionError.DuplicateValue;
}
```


maybe, it makes sense to return something like an enum which will say what, actually, was done inside the `insert`, either a fresh insertion or overwrite of the previously stored one.

what to you think?

Reply:
I think it would make more sense to simply return void instead of bool if this would only ever return false on error; using an enum can be elegant, but it then also then misses out on the type safety that the compiler provides in not allowing the caller to ignore errors in the same way as a simple enum value.

Reply:
sorry, I wasn't clear enough, I meant something like `!enum`

Reply:
nice thats a great idea - [ebfee79](https://github.com/Syndica/sig/pull/276/commits/ebfee79d6f7f4179d00d0be32bfcb66791af80e0)

Reply:
TBH, I thought about something simpler, like [file_ensure_content](https://github.com/tigerbeetle/tigerbeetle/blob/main/src/shell.zig#L275) function in TigerBeetle's codebase.

But your solution I like more because it gives more flexibility and an easy way to track and clean up old/duplicate values 💪